scripts/us_census/pep/monthly_population_estimate/preprocess.py

```diff
@@ -0,0 +1,385 @@
+        df['Date'] = df['Year'] + df['Month']
+        df['Date'] = df['Date'].str.replace(".", "").str.replace(
+            " ", "").str.replace("*", "")
+        df['Date'] = pd.to_datetime(df['Date'],
+                                    format='%Y%B%d',
+                                    errors="coerce")
+        df.dropna(subset=['Date'], inplace=True)
+        df['Date'] = df['Date'].dt.strftime('%Y-%m')
+        df.drop_duplicates(subset=['Date'], inplace=True)
+
+        #Deriving new SV Count_Person_InArmedForcesOverseas as
+        #subtracting Resident Population from
+        #Resident Population Plus Armed Forces Overseas
+        df['Count_Person_InUSArmedForcesOverseas'] = df[
+            'Resident Population Plus Armed Forces Overseas'].astype(
+                'int') - df['Resident Population'].astype('int')
+        computed_cols = ["Date", "Count_Person_InUSArmedForcesOverseas"]
+
+        #Selecting Coumputed and Final Columns from the DF.
+        df = df[computed_cols + final_cols]
+
+        #Renaming DF Headers with ref to SV's Naming Standards.
+        final_cols_list = ["Count_Person_" + col\
+                        .replace("Population ", "")\
+                        .replace("Population", "")\
+                        .replace(" Plus ", "Or")\
+                        .replace("Armed Forces Overseas", \
+                            "InUSArmedForcesOverseas")\
+                        .replace("Household", \
+                                "ResidesInHousehold")\
+                        .replace("Resident", "USResident")\
+                        .replace("Noninstitutionalized", \
+                            "NonInstitutionalized")\
+                        .strip()\
+                        .replace(" ", "_")\
+                        .replace("__", "_")\
+                        for col in final_cols]
+
+        final_cols_list = computed_cols + final_cols_list
+        df.columns = final_cols_list
+
+        #Multiplying the data with scaling factor.
+        for col in final_cols_list:
+            if "count" in col.lower():
+                if self.scaling_factor != 1:
+                    df[col] = df[col].astype('float', errors="ignore").multiply(
+                        self.scaling_factor)
+                df[col] = df[col].astype('Int64', errors="ignore")
+        self.scaling_factor = 1
+
+        #Creating Location column with default value country/USA.
+        #as the dataset is all about USA country level only.
+        df.insert(1, "Location", "country/USA", True)
+        df.insert(0, 'date_range', year_range(df['Date']), True)
+        #print(df.head())
+        return df
+
+    def _transform_data(self, file, df):
+        """
+        This method calls the required functions to transform
+        the dataframe and saves the final cleaned data in
+        CSV file format.
+        """
+        path = ""
+
+        #Finding the file name
+        if "\\" in self.cleaned_csv_file_path:

```

Comment:
why not use `os.path.basename` ?

Reply:
Sure Prasanth, used OS module to get file name, dir name.